src/mono/mono/metadata/metadata.c

```diff
@@ -5927,13 +5935,53 @@ signature_equiv (MonoMethodSignature *sig1, MonoMethodSignature *sig2, int equiv
 		/* if (p1->attrs != p2->attrs)
 			return FALSE;
 		*/
-		if (!do_mono_metadata_type_equal (p1, p2, TRUE))
+		if (!do_mono_metadata_type_equal (p1, p2, TRUE, equiv_flags == SIG_EQUIV_FLAG_IGNORE_CMODS))
```

Comment:
I'm not really reviewing this in a lot of detail yet.  but the idea with the equiv flags is that they're a bitmask.  So this should do `(equiv_flags & SIG_EQIIV_FLAG_IGNORE_CMODS) != 0`

Reply:
Fixed